src/classes/potentialSet.cpp

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2024 Team Dissolve and contributors
+
+#include "classes/potentialSet.h"
+#include "base/lineParser.h"
+#include "classes/atomType.h"
+#include "classes/box.h"
+#include "classes/configuration.h"
+#include "io/export/data1D.h"
+#include "items/deserialisers.h"
+#include "items/serialisers.h"
+#include "templates/algorithms.h"
+
+PotentialSet::PotentialSet() { fingerprint_ = "NO_FINGERPRINT"; }
+
+PotentialSet::~PotentialSet() { potentials_.clear(); }
+
+// Reset partial arrays
+void PotentialSet::reset()
+{
+    potentials_.clear();
+    fingerprint_ = "NO_FINGERPRINT";
+}
+
+// Set new fingerprint
+void PotentialSet::setFingerprint(std::string_view fingerprint) { fingerprint_ = fingerprint; }
+
+// Return full map of potentials specified
+std::map<std::string, PotentialSet::EPData> &PotentialSet::potentialMap() { return potentials_; }
+const std::map<std::string, PotentialSet::EPData> &PotentialSet::potentialMap() const { return potentials_; }
+
+/*
+ * Operators
+ */
+
+void PotentialSet::operator+=(const double delta)
+{
+    for (auto &[key, potential] : potentials_)
+    {
+        potential.ep += delta;
+    }
+}
+
+void PotentialSet::operator+=(const PotentialSet &source)
+{
+    for (auto &[key, potential] : source.potentialMap())
+    {
+        potentials_[key].ep += potential.ep;
+    }
+}
+
+void PotentialSet::operator*=(const double factor)
+{
+    for (auto &[key, potential] : potentials_)
+    {
+        potential.ep *= factor;
+    }
+}
```

Comment:
```suggestion
PotentialSet& PotentialSet::operator+=(const double delta)
{
    for (auto &[key, potential] : potentials_)
        potential.ep += delta;
    return (*this);
}

PotentialSet& PotentialSet::operator+=(const PotentialSet &source)
{
    for (auto &[key, potential] : source.potentialMap())
        potentials_[key].ep += potential.ep;
    return (*this);
}

PotentialSet& PotentialSet::operator*=(const double factor)
{
    for (auto &[key, potential] : potentials_)
        potential.ep *= factor;
    return (*this);
}
```

A couple of changes here.  First, we're dropping the extra braces on the single line `for` loop.  I'm also having the operator return a reference to the value.  This is generally good practice for the assignment operators because it enables chaining.  For example,

```
  a += b += c
```

Honestly, if someone ever submitted code like that, it's not going to get through code review.  Still, I think that the compiler is a little happier if that is available, so we should leave it in.